mautic/mautic
# Remove unnecessary queries to integration entity table

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | No
| New feature/enhancement? (use the a.x branch)      | Yes
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | No
| Automated tests included?              | Yes
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
Stop executing unnecessary queries to the integration_entity table on add/update lead if lead sync related plugins are not enabled.

https://github.com/mautic/mautic/blob/5.x/app/bundles/PluginBundle/EventListener/LeadSubscriber.php#L61
https://github.com/mautic/mautic/blob/5.x/app/bundles/PluginBundle/Entity/IntegrationEntityRepository.php#L516



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Disable all lead sync related plugin
3. Unnecessary delete/update queries to the integration_entity table on add/update lead should not be executed.
